server/models/competition_instance.js

```diff
@@ -70,21 +90,24 @@ class CompetitionInstance {
     return this.hasStarted() ? this.comp.getWinners() : null;
   }
 
-  _createNewGame(gameInfo) {
+  _createNewGame(gameInfo, lastGame) {
     if (gameInfo) {
-      let gameId = this.gameEngine.create(gameInfo.gameParams);
-      let game = this.currentGame = this.gameEngine.get(gameId);
+      let lastGameState = lastGame ? { lastGame: _.omit(lastGame.game, "params") } : {};
```

Comment:
Yeah, this is a very good idea, I didn't remember that new games can depend on the initial parameters or even the result of the previous games.

Why did you opt to remove the `params` from the last game? So that you didn't end with a chain of game parameters, each one referencing the previous one?